silkworm/db/snapshots/bittorrent/web_seed_client.hpp

```diff
@@ -0,0 +1,82 @@
+/*
+   Copyright 2024 The Silkworm Authors
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+#pragma once
+
+#include <memory>
+#include <optional>
+#include <set>
+#include <string>
+#include <vector>
+
+#include <boost/asio/io_context.hpp>
+#include <boost/url/url.hpp>
+// Disable warnings raised during compilation of libtorrent
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wc++11-compat"
+#pragma GCC diagnostic ignored "-Wshadow"
+#pragma GCC diagnostic ignored "-Wsign-conversion"
+#include <libtorrent/torrent_info.hpp>
```

Comment:
is there a way to hide these inside third_party/libtorrent/CMakeLists.txt ?

Reply:
I would prefer to address this issue in another PR, because the same directives to disable warnings are present also on other pre-existing header files, e.g. [here](https://github.com/erigontech/silkworm/blob/master/silkworm/db/snapshots/bittorrent/client.hpp#L34)